### PR TITLE
Fix type docs for Notices

### PR DESCRIPTION
**Pull request by simono**
## Description
Fix documentation errors as explained in #17203.

## Types of changes
- fixes docs

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code follows the accessibility standards. 
- [x] My code has proper inline documentation. 
- [x] I've included developer documentation if appropriate.

### CODE REVIEW

File: packages/notices/src/store/selectors.js [hunk 1]
```diff
@@ -29,6 +29,8 @@ const DEFAULT_NOTICES = [];
  *                                      removal without notice.
  * @property {boolean} isDismissible    Whether the notice can be dismissed by
  *                                      user. Defaults to `true`.
+ * @property {string}  type             Type of notice, one of `default`,
+ *                                      or `snackbar`. Defaults to `default`.
  * @property {WPNoticeAction[]} actions User actions to present with notice.
```
Comment (ocean90) [inline packages/notices/src/store/selectors.js]: Just noticed that `speak` is missing as well. Can you please add as part of this PR?

**Review by ocean90 — CHANGES_REQUESTED** [review]

**Review by ocean90 — APPROVED** [review]

**Review by dsifford — APPROVED** [review]
See comments.

File: packages/notices/src/store/actions.js [hunk 2]
```diff
@@ -23,6 +23,9 @@ import { DEFAULT_CONTEXT, DEFAULT_STATUS } from './constants';
  * @param {?boolean}               options.isDismissible Whether the notice can
  *                                                       be dismissed by user.
  *                                                       Defaults to `true`.
+ * @param {?string}                options.type          Type of notice, one of
```
[the remark this answers is not in the document]
Comment (simono) [reply packages/notices/src/store/actions.js]: From what I understand it's nullable and gets a default value if it's null, see https://github.com/WordPress/gutenberg/blob/132d2d4cf1b24511d821e6e1fe2203fb4c18d201/packages/notices/src/store/actions.js#L38

Or did I miss something?

[the remark this answers is not in the document]
Comment (simono) [reply packages/notices/src/store/actions.js]: Ok, thanks for the clarification :)

I can update this PR and change the params from nullable to optional with default. Would that be helpful @dsifford?

[the remark this answers is not in the document]
Comment (simono) [reply packages/notices/src/store/actions.js]: I've updated the PR and fixed the params in `notices/src/store/actions.js`.

**Comment by gziolo** [comment]
Travis job fails. It looks like you need to run “npm run docs:build” to refresh docs.

**Comment by simono** [comment]
Thanks for the heads-up @gziolo 
The build passed now.

Just out of curiosity: Why are those generated docs in version control?

**Comment by gziolo** [comment]
> Just out of curiosity: Why are those generated docs in version control?

We use GitHub source to power developer docs: https://developer.wordpress.org/block-editor/data/data-core-notices/ but at the same time, you can browse them on GitHub. Ideally, bot should run `npm run docs:build` and add a commit to your PR when necessary. I hope we will get there one day :)

**Comment by gziolo** [comment]
Congrats on your first code contribution to Gutenberg 🎉

**Comment by simono** [comment]
Thanks for the merge and all the help @gziolo

**Pull request merged**